### DISCUSSION TITLE
treatment of deletion confirmation on top of side modal

### DISCUSSION THREAD

**Discussion by simsemma** [twilio-labs/paste#3209] [General] [answered]
currently working on a delete confirmation dialog when a user clicks to remove an item in the side modal. is it ok to throw a confirmation dialog on top of the side modal, or is there another recommended treatment? i feel like it's typically not great to do a dialog on top of a side modal here, but i feel like it might be the only treatment to get at this use case.

![image](https://user-images.githubusercontent.com/84725143/235221304-bc078593-4272-42a9-a783-67d9124d3a98.png)

there's also another instance where we do this currently within the segment app (it's not a delete, but it's a confirmation of navigating away from the side sheet, so a bit different) when you're on the side sheet: if a customer tries to navigate away from the support ticket they're filling out, we show this:
<img width="1475" alt="Screen Shot 2023-04-28 at 2 06 43 PM" src="https://user-images.githubusercontent.com/84725143/235221572-39675852-9c0c-4b2f-8c83-37d396870f6f.png">

**Comment by serifluous** [comment]
Hey @simsemma ! Can you explain more about "i feel like it's typically not great to do a dialog on top of a side modal here"? Is it the asymmetry or the overlay-on-overlay (or something else) part of the treatment that makes you feel like it's not great?

**Reply by simsemma** [reply]
the overlay-on-overlay 😄

**Reply by serifluous** [reply, marked as answer]
Yeeeah, it's one of the downsides of over-relying on overlays and dialogs 😬, but there's nothing necessarily wrong with it. Alert Dialogs are meant to focus attention and block everything else, so they're serving their purpose here.

If the [action is low-severity enough](https://paste.twilio.design/patterns/delete#low-severity) / pretty easily reversible and you're trying to avoid the overlay-on-overlay, you could consider not showing the Alert Dialog on-delete, and [pop a toast when the deletion has gone through, ideally with an undo button](https://paste.twilio.design/patterns/delete#post-deletion).

**Reply by simsemma** [reply]
ah ok thanks, that's super helpful! i think in this case it's more than low-severity, so perhaps relying on the dialog is the right way to go 🤔
